### ISSUE TITLE
Add label encoder to XGBoostClassifier

### ISSUE THREAD

**Issue by angela97lin** [alteryx/evalml#2517]
I currently get the following DeprecationWarning with XGBoost==1.4.2:
```
  The use of label encoder in XGBClassifier is deprecated and will be removed in a future release. To remove this warning, do the following: 1) Pass option use_label_encoder=False when constructing XGBClassifier object; and 2) Encode your labels (y) as integers starting with 0, i.e. 0, 1, 2, ..., [num_class - 1].
```
Since I know we want to upgrade XGBoost to >1.4.2 for the pandas 1.3.0 release, it could be a good idea to address this too, sooner rather than later :)

**Comment by kadatatlukishore** [comment]
Hello @angela97lin , Can you please guide me to solve this issue ?

**Comment by angela97lin** [comment]
Hi @kadatatlukishore, sure thing! I think there are a few things that would need to be done:

1) Update [XGBoostClassifer](https://github.com/alteryx/evalml/blob/main/evalml/pipelines/components/estimators/classifiers/xgboost_classifier.py) to pass to the `xgb.XGBClassifier(random_state=random_seed, **parameters)` instance `use_label_encoder=False`
2) Add a label encoder to the `XGBoostClassifier` object. I think one great place to look at the [LightGBMClassifier](https://github.com/alteryx/evalml/blob/main/evalml/pipelines/components/estimators/classifiers/lightgbm_classifier.py) which already has a pattern in place for the label encoder.
3) Write unit tests where the target is not numeric, and ensure that `fit`, `predict`, and `predict_proba` work as expected 😁

**Comment by kadatatlukishore** [comment]
Thanks @angela97lin ! Will start working on this.

**Comment by angela97lin** [comment]
@kadatatlukishore Sounds good, let me know if you need any help or clarifications! 😁

**Comment by kadatatlukishore** [comment]
Hello @angela97lin , Is this fine ?

```bash
       self._label_encoder = None
        xgb_classifier = xgb.XGBClassifier(random_state=random_seed,  use_label_encoder=False, **parameters)
        super().__init__(
            parameters=parameters, component_obj=xgb_classifier, random_seed=random_seed
        )

    def fit(self, X, y=None):
        X, y = super()._manage_woodwork(X, y)
        self.input_feature_names = list(X.columns)
        X = _rename_column_names_to_numeric(X, flatten_tuples=False)
        if not is_integer_dtype(y):   # to check if the "y" is numeric or not
            self._label_encoder = LabelEncoder()   # declaring the LabelEncoder
            y = pd.Series(
                self._label_encoder.fit_transform(y), dtype="int64"
            ) # transformation 
        self._component_obj.fit(X, y)
        return self
```

**Comment by kadatatlukishore** [comment]
Please review this https://github.com/kadatatlukishore/Eval-ml/tree/main   @angela97lin  , 
I've changed the code and started testing it according, I've not installed evalml through pip. I think it's working but not sure if the expected output would be different or it will be the same.

**Comment by angela97lin** [comment]
Hi @kadatatlukishore, I think you're off to a great start!

You'll also need to update `predict` to handle label encoding too! If we encode during fit, the estimator will be expecting label-encoded data and will return such in `predict`. We need to do the inverse-transformation before we return to the user.

Concrete example:
target = ["a", "b"]
XGB will label encode during fit as [1, 2]
`predict` will now predict [1,2], but if a user gets [1,2] they might be confused! We want to return the original target values, so we need to do an inverse transformation to remap [1,2] to ["a", "b"]. 

Does that make sense?

**Comment by kadatatlukishore** [comment]
Yes clearly understood, thanks @angela97lin . Will make changes to the predict function.

**Comment by kadatatlukishore** [comment]
```bash
  def predict(self, X):
      X = _rename_column_names_to_numeric(X, flatten_tuples=False)
      y_pred = super().predict(X)
      if self._label_encoder:
          y_pred = self._label_encoder.inverse_transform(np.array(y_pred))
          
      return y_pred
```
I've changed the predict function. Is it fine @angela97lin ?

**Comment by angela97lin** [comment]
@kadatatlukishore Sure! :) For next steps, I'd suggest putting up a PR, writing unit tests to make sure that everything is working as expected, and then requesting review from the EvalML team so that we can take a look 😁

**Comment by kadatatlukishore** [comment]
Thank you so much @angela97lin

**Comment by angela97lin** [comment]
No problem, always excited to have new contributors!

**Comment by kadatatlukishore** [comment]
Hello @angela97lin , can you please help me in unit testing and linting up. I am trying but not sure I understand this correctly, as a newbie it's hard for me to get understand this.

**Comment by angela97lin** [comment]
Hi @kadatatlukishore, I heard from @freddyaboulton that you guys are working together--that's great to hear! If there's anything else you need, we can continue the thread on your PR :)

**Comment by kadatatlukishore** [comment]
Sure means a lot, Thanks @angela97lin
